### ISSUE TITLE
Error importing RNN Model

### ISSUE THREAD

**Issue by ETTAN93** [unit8co/darts#2431]
Hi,

Two questions on my side:

**1.  Importing RNN Model**
I have upgraded my Darts version to the latest, i.e. 0.30.0
![image](https://github.com/unit8co/darts/assets/59031005/cd517981-6f84-4ac6-aad9-cd251168ff0f)

When trying to import RNNModel, one of the underlying imports seem to not be working.
![image](https://github.com/unit8co/darts/assets/59031005/8c4d42ea-a511-45f9-b488-b9827ef8e70d)

Am I importing something incorrectly?

2. [**Example RNN Notebook**](https://github.com/unit8co/darts/blob/master/examples/08-DeepAR-examples.ipynb)

Looking at the example in github, I noticed that the dataset was first split into a train and test set (called val_en) in the notebook. The RNN model is then fitted on the train set and validated on the test set. The best performing model is then loaded and used to do a historical forecast with the same train and test split. Would this not result in inaccurate metrics (data leakage) since the test set used to fit the model and then used to run historical forecast is the same test set?

**Comment by dennisbader** [comment]
From a first glance, it looks to me like something went wrong in the installation. Can you try re-installing darts from scratch in a new environment?

**Comment by ETTAN93** [comment]
@dennisbader thanks. uninstallinng and reinstalling with u8darts[all] seemed to have solved the problem. 

Looking at the [example notebook](https://github.com/unit8co/darts/blob/master/examples/08-DeepAR-examples.ipynb) for deepAR in github, I noticed that the dataset was first split into a train and test set (called val_en) in the notebook. The RNN model is then fitted on the train set and validated on the test set. The best performing model is then loaded and used to do a historical forecast with the same train and test split. Would this not result in inaccurate metrics (data leakage) since the test set used to fit the model and then used to run historical forecast is the same test set?

**Comment by madtoinou** [comment]
Hi @ETTAN93,

Good to see that your initial problem was solved.

You're right, running historical forecast on the validation set is not ideal and the initial series should have been split into 3 sets; train, validation & test. The model will probably over-perform in this particular setup since we already selected the best performing model based on the validation set during training (to avoid over-fitting) but `backtest()` is still informative since it indicates the performance of the model on each forecast horizon.

Closing the issue since the original problem was solved.
